lib/datadog/core/backport.rb

```diff
@@ -21,5 +21,25 @@ def string_delete_prefix(string, prefix)
         end
       end
     end
+
+    # This module is used to provide features from Ruby 2.4+ to older Rubies
+    module BackportFrom24
+      module_function
+
+      if RUBY_VERSION < '2.4'
+        def dup(value)
```

Comment:
So about using `module_function` and `dup` here... 

Using `module_function` means that we can use the helpers from this module directly (e.g. `BackportFrom24.dup`) or by including it in a class. But with the current naming, you'd be overwriting `Kernel#dup`.

Aka either:
* Name this `kernel_dup(value)` (or `object_dup`?) so that including it as a class from other module doesn't have disastrous effects
* Don't make this a `module_function`, just a good ole `self.dup`

Now since I've been annoying you in this PR from the beginning in this PR about explicitness, you'll probably not be very surprised if I tell you I'm not a fan of module function and that I prefer that in some code we do 
`Core::BackportFrom24.dup(...)` rather than some mysterious `kernel_dup`, but I'll leave the decision to you :)

(I'll add that yeah the other `BackportFrom25` also uses `module_function`, but it never actually gets included anywhere, so it's a tiny diff to also convert it to sue `self` :P)